CtrlVAF/CtrlVAF/Events/EventDispatcher.cs

```diff
@@ -31,13 +34,26 @@ public class EventDispatcher : Dispatcher
         /// <param name="command">A command of type TCommand that has inherited from <see cref="Commands.IEventCommand{T}"/></param>
         /// <param name="throwExceptions">Whether or not to stop executing ICommandHandlers upon exceptions and throw the exception</param>
         /// <param name="exceptionHandler">An exception handler to pass along or handle any ICommandHandler exceptions</param>
-        public EventDispatcher()
+        public EventDispatcher(ConfigurableVaultApplicationBase<TConfig> vaultApplication)
         {
+            this.vaultApplication = vaultApplication;
         }
 
         /// <inheritdoc/>
         public override void Dispatch(params ICtrlVAFCommand[] commands)
         {
+            var wrongCommands = commands.ToList().RemoveAll(cmd =>
```

Comment:
Bear in mind that the call to RemoveAll on the list does not modify the commands passed to the dispatch command. Meaning, that any "wrong commands" are still passed along to be handled. But maybe this is intentional?

Reply:
It doesn't? I'm fairly sure it does. tbf my variable names are quite horrid here.

But RemoveAll should remove the commands from the list and return how many were removed as an integer.

Reply:
Not in the way that it is implemented here. Take this example for example:
```csharp
            IEnumerable<int> commands = new List<int>() { 1, 2, 3, 4 };
            var removed = commands.ToList().RemoveAll(i => i == 2);
            IEnumerable<int> expected = new List<int>() { 1, 3, 4 };

            Assert.IsTrue(removed == 1);
            Assert.IsTrue(expected.SequenceEqual(commands));
```
When calling "ToList()" on the commands collection, you are creating a separate clone of the collection. This means that all modifications are applied to the clone, and not the commands IEnumerable itself. The test above will fail as a result.

If you instead consider this test:
```csharp
            IEnumerable<int> commands = new List<int>() { 1, 2, 3, 4 };
            var commandList = commands.ToList();
            var removed = commandList.RemoveAll(i => i == 2);
            IEnumerable<int> expected = new List<int>() { 1, 3, 4 };

            Assert.IsTrue(removed == 1);
            Assert.IsTrue(expected.SequenceEqual(commandList));
```
The RemoveAll method is called on the same list as is checked in the end.

The main issue with your code is that GetTypes and HandleConcreteTypes in your code uses the unmodified reference to the commands, and not the modified one that you do not save after making it a list :)

Reply:
Ah you're right. Thanks 😄 I've changed the filter to something easier.